AssemblerLib/MeshComponentMap.cpp

```diff
@@ -55,12 +69,23 @@ MeshComponentMap::MeshComponentMap(
                 // mesh items are ordered first by node, cell, ....
                 for (std::size_t j=0; j<mesh_subset.getNNodes(); j++)
                 {
-                    const GlobalIndexType global_id = static_cast<GlobalIndexType>(components.size()
+                    GlobalIndexType global_id = static_cast<GlobalIndexType>(components.size()
                                                * mesh.getGlobalNodeID(j) + comp_id);
-                    const GlobalIndexType signed_global_id = mesh.isGhostNode( mesh.getNode(j)->getID() )
-                                                    ? -global_id : global_id;
+                    const bool is_ghost = mesh.isGhostNode( mesh.getNode(j)->getID() );
+                    if (is_ghost)
+                    {
+                        _ghosts_indices.push_back(global_id);
```

Comment:
The values in ``_ghost_indices`` are all positive. The next line set the global index negative for the equation assembly.